### PR TITLE
Taboo: Add TurnManager (skeleton) + Turn (message generation)

### PR DESCRIPTION
**Pull request by sbirmi**
- TurnManager and Turn aren't instantiated anywhere in the Room yet.
- Added test for messages generated from Turn

### CODE REVIEW

File: src/Taboo/Turn.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                                    # the dictionary in msg1)
+            msg2[3].update({
+                "secret": self.secret,
+                "disallowed": self.disallowed,
+            })
+            self.privateMsgSrc.setMsgs([Jmai(msg2, None)])
+
+            return
+
+        # If the word isn't in play, there is no private messaging needed.
```
Comment (lambdaFooBar) [inline src/Taboo/Turn.py:70]: "word" = "turn"

File: src/Taboo/Turn.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        self.score = score or [] # list of teamNumbers that should be awarded points
+
+        # Messages are broadcast to everyone connected to the room
+        self.publicMsgSrc = MsgSrc(self.allConns)
+
+        # If we are in play, we have to send private messages revealing
+        # The word in play to some people
+        self.privateMsgSrc = None
+        if state == TurnState.IN_PLAY:
+            self.privateConnsGrp = ConnectionsGroup()
```
Comment (lambdaFooBar) [inline src/Taboo/Turn.py:40]: do we need to make it a class-attribute?  Can it not be local?

File: src/Taboo/Turn.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                 player, otherTeams, allConns,
+                 state=TurnState.IN_PLAY,
+                 score=None):
+        self.turnIdx = turnIdx
+        self.wordIdx = wordIdx
+        self.secret = secret
+        self.disallowed = disallowed
+        self.player = player
+        self.otherTeams = otherTeams
+        self.allConns = allConns
```
Comment (lambdaFooBar) [inline src/Taboo/Turn.py:29]: do we need to make it a class-attribute?

**Review by lambdaFooBar — APPROVED** [review]

File: src/Taboo/Turn.py (hunk 2 above)
Comment (sbirmi) [reply src/Taboo/Turn.py:40]: this can't be local (we discussed offline)

**Pull request merged**